src/client/datascience/jupyter/jupyterSession.ts

```diff
@@ -196,11 +210,17 @@ export class JupyterSession extends BaseJupyterSession {
     private async createSession(
         serverSettings: ServerConnection.ISettings,
         kernelConnection: KernelConnectionMetadata | undefined,
-        cancelToken?: CancellationToken
+        cancelToken?: CancellationToken,
+        disableUI?: boolean
     ): Promise<ISessionWithSocket> {
         // Create our backing file for the notebook
         const backingFile = await this.createBackingFile();
 
+        // Make sure the kernel has ipykernel installed if on a local machine.
+        if (kernelConnection?.interpreter && this.connInfo.localLaunch) {
```

Comment:
This spot here is where we'll install ipykernel if necessary just before starting a kernel.

I believe in the future this will also be where we'll register a kernel (or maybe not as I don't think it's actually necessary)